cg/meta/store/balsamic.py

```diff
@@ -3,36 +3,38 @@
 import logging
 import os
 import shutil
+from pathlib import Path
 
 import ruamel.yaml
 from cg.exc import AnalysisDuplicationError
-from pathlib import Path
 
 LOG = logging.getLogger(__name__)
 
 
-def gather_files_and_bundle_in_housekeeper(config_stream, hk_api, status, case_obj):
-    """Function to gather files and bundle in housekeeper"""
-
-    bundle_data = _add_analysis(config_stream, case_obj)
+def gather_files_and_bundle_in_housekeeper(
+    config_path, deliverables_file, hk_api, status, case_obj
+):
+    """Function to create a bundle for an analysis in housekeeper"""
 
+    bundle_data = _parse_bundle_data(config_path, deliverables_file, case_obj)
     results = hk_api.add_bundle(bundle_data)
+
     if not results:
-        raise AnalysisDuplicationError("analysis version already added")
+        raise AnalysisDuplicationError("Analysis version already added")
     bundle_obj, version_obj = results
 
     _reset_analysis_action(case_obj)
     new_analysis = _create_analysis(bundle_data, case_obj, status, version_obj)
-    version_date = version_obj.created_at.date()
-    LOG.info(f"new bundle added: {bundle_obj.name}, version {version_date}")
     _include_files_in_housekeeper(bundle_obj, hk_api, version_obj)
+
     return new_analysis
 
 
 def _include_files_in_housekeeper(bundle_obj, hk_api, version_obj):
```

Comment:
This is not really balsamic specific, could be better to have in HK api so everyone is using this. I think that this is one big problem in CG, that the same thing is done in multiple ways all through the code